src/hotspot/share/utilities/utf8.cpp

```diff
@@ -518,7 +518,7 @@ void UNICODE::as_quoted_ascii(const T* base, int length, char* buf, int buflen)
       *p++ = (char)c;
     } else {
       if (p + 6 >= end) break;      // string is truncated
-      sprintf(p, "\\u%04x", c);
+      os::snprintf(p, 7, "\\u%04x", c);
```

Comment:
This should be 6, or? We have 6 characters left before end, assuming end is exclusive.

Also, maybe use a named constant?

Reply:
For snprintf, all bytes written to the buffer (including the terminating \0) are counted. You have 6 bytes for character encoding ("\uxxxx") and one byte for \0. Code is correct.